python-sdk/nuscenes/predict/helper.py

```diff
@@ -0,0 +1,181 @@
+
+from typing import Dict, Tuple, Any, List
+from pyquaternion import Quaternion
+from nuscenes import NuScenes
+from nuscenes.utils.geometry_utils import transform_matrix
+from nuscenes.eval.common.utils import quaternion_yaw
+import numpy as np
+
+MICROSECONDS_PER_SECOND = 1e6
+BUFFER = 0.15 # seconds
+
+def angle_of_rotation(yaw: float) -> float:
+    return (np.pi / 2) + np.sign(-yaw) * np.abs(yaw)
+
+def convert_global_coords_to_local(coordinates: np.ndarray,
+                                   translation: Tuple[float, float, float],
+                                   rotation: Tuple[float, float, float, float]) -> np.ndarray:
+    """Converts global coordinates to coordinates in the frame given by the rotation quaternion and
+    centered at the translation vector.
+    :param coordinates: x,y locations. array of shape [n_steps, 2]
+    :param translation: Tuple of (x, y, z) location that is the center of the new frame
+    :param rotation: Tuple representation of quaternion of new frame."""
```

Comment:
@freddyaboulton Do we have a clearly defined representation for a quaternion? There are two major conventions.

Reply:
Also, this is just for rotations around the z axis, no? Should be clear that it is not a general rotation routine.

Reply:
Added a bit of info on the quaternion representation but it is the same as the one stored in the nuScenes dataset.